### ISSUE TITLE
FlexPaper_1.3rc1 exception: The supplied DisplayObject must be a child of the caller

### ISSUE THREAD

**Issue by GoogleCodeExporter** [fleszar1/flexpaper#82]
```
Hello,

What steps will reproduce the problem?

- Load flexpaper (ver 1.3rc1) in browser (FireFox or IE) with any
content SWF file.

What is the expected output? What do you see instead?

Following exception is thrown by flash player (inconsistently):

ArgumentError: Error #2025: The supplied DisplayObject must be a child of
the caller.
    at flash.display::Loader/_loadBytes()
    at flash.display::Loader/loadBytes()
    at
com.devaldi.streaming::AVM2Loader/confirmBytesLoaded()[/Users/Erik/Documents/Ado
be
Flash Builder 4/FlexPaper/src/com/devaldi/streaming/AVM2Loader.as:132]
    at Function/http://adobe.com/AS3/2006/builtin::apply()
    at <anonymous>()
    at SetIntervalTimer/onTimer()
    at flash.utils::Timer/_timerDispatch()
    at flash.utils::Timer/tick()

What version of the product are you using? On what operating system?
- FlexPaper_1.3rc1 on WinXP

Additional Info:

Did search on google about this exception and found following pointers:

http://www.actionscript.org/forums/showpost.php3?p=696785&postcount=9

(full post for above URL)
http://www.actionscript.org/forums/showthread.php3?t=138634

Thanks,

-Darshan

```

Original issue reported on code.google.com by `darshanm...@gmail.com` on 27 May 2010 at 10:10

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the report and the information about this issue. I will make sure 
this gets 
addressed in the final release of 1.3
```

Original comment by `erik.eng...@gmail.com` on 27 May 2010 at 8:14
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
more update on this:

- This error stops if ProgressiveLoading is set to 'false'

- Also, further search about this on forums indicates that
the error seems to be related to removing of non-existent
child object which is happening inside loadBytes() 
(AVM2Loader.confirmBytesLoaded())
.I could reproduce same error by calling _loader.unload() just before
loadBytes() in confirmBytesLoaded(). More info at: 
http://www.senocular.com/flash/tutorials/runtimeerrors/

- For now, I have wrapped this error causing code in confirmBytesLoaded()
inside try/catch block with trace inside catch to get notification. The viewer
fails to load remaining pages after this error is thrown. 

-darshan




```

Original comment by `darshanm...@gmail.com` on 29 May 2010 at 2:09

**Comment by GoogleCodeExporter** [comment]
```
Hi, actually having a problem to reproduce this... starting to wonder if its 
about the content-length on your web 
server not getting returned and this resulting as a consequence... what flash 
version on firefox/ie are you 
running?
```

Original comment by `erik.eng...@gmail.com` on 29 May 2010 at 11:27

**Comment by GoogleCodeExporter** [comment]
```
Please try the attached build and see if you can reproduce the issue. Have made 
some adjustments to how the 
last part of the swf is loaded


```

Original comment by `erik.eng...@gmail.com` on 29 May 2010 at 11:36

Attachments:
- [FlexPaperViewer.swf](https://storage.googleapis.com/google-code-attachments/flexpaper/issue-82/comment-4/FlexPaperViewer.swf)

**Comment by GoogleCodeExporter** [comment]
```
I am using Apache 2.2 on local Win XP. Firefox is ver 3.6.2

Will try your build and let you know.

-Darshan

```

Original comment by `darshanm...@gmail.com` on 30 May 2010 at 5:33

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Hi Erik,

The error condition was reproduced with the FlexPaperViewer.swf you 
had attached. I am attaching a screenshot or the error in browser.

Web Server: Apache 2.2 on localhost (WinXP SP2)
Browser: Firefox 3.6.2, with Flash Player plug-in ver 9.0.124.0

-darshan
```

Original comment by `darshanm...@gmail.com` on 30 May 2010 at 6:41

Attachments:
- [issue82.png](https://storage.googleapis.com/google-code-attachments/flexpaper/issue-82/comment-7/issue82.png)

**Comment by GoogleCodeExporter** [comment]
```
This must be an old flash player 9 runtime bug that has been fixed in version 
10 and above because I can't 
reproduce it in version 10. I'll do some testing in v9 and tell you how it goes
```

Original comment by `erik.eng...@gmail.com` on 30 May 2010 at 7:08

**Comment by GoogleCodeExporter** [comment]
```
I was able to reproduce this in Flash Player 9 and I think I've found a 
solution to it by basically retrying the 
operation that fails when it fails.. Have not been able to reproduce the issue 
in this build. Thanks for your help in 
finding this issue. Please tell me how it goes. 

Please see attached. 
```

Original comment by `erik.eng...@gmail.com` on 30 May 2010 at 9:48

Attachments:
- [FlexPaperViewer.swf](https://storage.googleapis.com/google-code-attachments/flexpaper/issue-82/comment-9/FlexPaperViewer.swf)

**Comment by GoogleCodeExporter** [comment]
```
Yes. It doesn't throw the exception now.
Just tried the swf you sent several times.

Thanks for all the efforts.

-darshan


```

Original comment by `darshanm...@gmail.com` on 30 May 2010 at 3:04

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.3 final & code repository
```

Original comment by `erik.eng...@gmail.com` on 30 May 2010 at 8:21
- Changed state: **Fixed**
